qt5: Add missing include path to pkg-config files

xbreak opened the pull request:
###### Motivation for this change

To fix #52457.

 This is achieved by patching qtbase `qmake/generators/makefile.cpp` to
    unconditionally add the missing `-I${includedir}`. The include path is
    otherwise conditioned on whether it is already available or not. Since there is
    no unified set of system include paths in nix this cause problems as reported
    in #52457.


<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->




###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested a _sample_ compilation of pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

pkg-config when this fix is applied to staging:
```
 $ grep "Cflags" /nix/store/szikc0g33cgh55ihkfwi983z0psw0ciz-qtdeclarative-5.12.7-dev/lib/pkgconfig/*
/nix/store/szikc0g33cgh55ihkfwi983z0psw0ciz-qtdeclarative-5.12.7-dev/lib/pkgconfig/Qt5Qml.pc:Cflags: -DQT_QML_LIB -I${includedir}/QtQml -I${includedir}
/nix/store/szikc0g33cgh55ihkfwi983z0psw0ciz-qtdeclarative-5.12.7-dev/lib/pkgconfig/Qt5Quick.pc:Cflags: -DQT_QUICK_LIB -I${includedir}/QtQuick -I${includedir}
/nix/store/szikc0g33cgh55ihkfwi983z0psw0ciz-qtdeclarative-5.12.7-dev/lib/pkgconfig/Qt5QuickTest.pc:Cflags: -DQT_QMLTEST_LIB -I${includedir}/QtQuickTest -I${includedir}
/nix/store/szikc0g33cgh55ihkfwi983z0psw0ciz-qtdeclarative-5.12.7-dev/lib/pkgconfig/Qt5QuickWidgets.pc:Cflags: -DQT_QUICKWIDGETS_LIB -I${includedir}/QtQuickWidgets -I${includedir}
```

jtojnar commented on pkgs/development/libraries/qt-5/qtModule.nix:
```diff
@@ -34,13 +34,15 @@ mkDerivation (args // {
                 -e "/^prefix=/ c prefix=''${!outputLib}" \
                 -e "/^exec_prefix=/ c exec_prefix=''${!outputBin}" \
-                -e "/^includedir=/ c includedir=''${!outputDev}/include"
+                -e "/^includedir=/ c includedir=''${!outputDev}/include" \
```
This is werid. Multiple outputs setup hook should already handle this:

https://github.com/NixOS/nixpkgs/blob/f3ce7c5582c1a71baf133e6e85e8517e162abc50/pkgs/build-support/setup-hooks/multiple-outputs.sh#L161-L164

xbreak replied:
That change in `qtModule.nix` is [3 years old](https://github.com/NixOS/nixpkgs/commit/fe0ab944db2176e6757ae89aeebea16ab8dc69d0#diff-88c589c484358b9563a1084ccf34fb69R43), so maybe its superfluous now. 

I made a quick test and the results does indeed seem fine without it. It's however unrelated to what I'm trying to accomplish.

xbreak commented:
@ttuegel: A better approach than do .pc file patching (as found by @jtojnar) might be to set an empty `QMAKE_DEFAULT_INCDIRS`. What do you think?

See discussion in #52457.

jtojnar commented:
It would be better to find out what is setting the `QMAKE_DEFAULT_INCDIRS` since otherwise we need to nullify it in every package using `qt_module.prf` (e.g. qtsvg).

xbreak committed:
qt5: Add missing include path to pkg-config files

This is achieved by patching qtbase `qmake/generators/makefile.cpp` to
unconditionally add the missing `-I${includedir}`. The include path is
otherwise conditioned on whether it is already available or not. Since there is
no unified set of system include paths in nix this cause problems as reported
in #52457.

xbreak commented:
@ttuegel: Updated PR as agreed in #52457.

jtojnar commented on pkgs/development/libraries/qt-5/5.12/default.nix:
```diff
@@ -67,6 +67,7 @@ let
         ./qtbase.patch.d/0010-qtbase-qtpluginpath.patch
         ./qtbase.patch.d/0011-qtbase-assert.patch
         ./qtbase.patch.d/0012-fix-header_module.patch
+        ./qtbase.patch.d/0014-qtbase-pkg-config.patch
```
We really need to start writing comments.

```suggestion

        # Ensure -I${includedir} in Cflags in pkg-config files.
        # See https://github.com/NixOS/nixpkgs/issues/52457
        ./qtbase.patch.d/0014-qtbase-pkg-config.patch

```

xbreak replied:
Agreed. PR is updated.

xbreak commented:
Ping @ttuegel:  Do you mind taking a look?
